### ISSUE TITLE
HLS on iOS broken

### ISSUE THREAD

**Issue by t-schroeder** [opencast/opencast#1452]
**Describe the bug**
iOS devices don't play HLS videos with PaellaPlayer. It doesn't matter which browser you use. When you click a video the load animation starts but never ends.

**To Reproduce**

1. Configure HLS as the first streaming protocol.
2. Either:
a. Take an actual iOS device.
b. Use any other OS and switch the browser's user agent to iOS:
`Mozilla/5.0 (iPhone; CPU iPhone OS 13_2_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Mobile/15E148 Safari/604.1`
3. Try to watch a video.
4. Inspect it and notice that you see something like this:
```
<video id=...>
    <source src=...m3u8" type="undefined">
</video>
```

**Environment Information**
 - OS: iOS 10, iOS 13
 - Browser: Safari, Chrome, Firefox
 - Opencast: 8.1

**Comment by t-schroeder** [comment]
If you replace type="undefined" with type="application/x-mpegURL" the video works again.

I debugged through the JS and found this place that seems to be the cause of this problem:

https://github.com/opencast/opencast/blob/9f1584c660c738e84ba26b4224cb6af1caa83c51/modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js#L137-L140

In case of HLS there is no track.video, but track.mimetype is "application/x-mpegURL".

This regression is caused by https://github.com/opencast/opencast/commit/5b3a0933833e9d03621dbcaf01d81d4773ec686c. So I'm tagging @miesgre here.

**Comment by miesgre** [comment]
@t-schroeder I submited a fix, but I can not test it because I do not have an iOS device to test it.
Can you test it?

**Comment by t-schroeder** [comment]
@miesgre Thanks for the quick fix. I tested it on an iOS device and it works. 👍

**Comment by lkiesow** [comment]
@t-schroeder, thanks for testing this. I've merged the patch so that it will be part of Opencast 8.2.

A request for next time, it's really helpful if you comment on the pull request rather than the issue if you test a patch since we as developers can then easily see that someone tested the patch and it can be safely merged. Chances are high that that's missed if the comment is on the issue. Anyway, I did notice it this time ;-)
